### PR TITLE
Grouping AMI channels together + multi-channel cut API improvements

### PR DESCRIPTION
**Pull request by pzelasko**
AMI can be a great testbed for multi-channel capabilities in Lhotse, so I'm making a few improvements in its preparation sript. The idea is that we can construct a CutSet with entries like this:

image

I'll still need to add tests so this is not finished.

### CODE REVIEW

**Review by danpovey — COMMENTED** [review]
small comments

File: lhotse/cut.py [hunk 1]
```diff
@@ -970,6 +971,29 @@ def trim_to_unsupervised_segments(self) -> 'CutSet':
...
         return CutSet.from_cuts(cuts)
 
+    def mix_same_recording_channels(self) -> 'CutSet':
+        """
+        Find cuts that come from the same recording and have matching start and end times, but
+        represent different channels. Then, mix them together (in matching groupbs) and return
+        a new ``CutSet`` that contains their mixes. This is useful for processing microphone array
+        recordings.
+
+        It is inteded to be used as the first operation after creating a new ``CutSet`` (but
```
[the remark this answers is not in the document]
Comment (pzelasko) [reply lhotse/cut.py:981]: ha, I guess it's easy to tell I'm writing this in the late hours by the number of typos ;)

File: lhotse/recipes/ami.py [hunk 2]
```diff
@@ -161,58 +169,70 @@ def prepare_ami(
...
-                            duration=duration,
-                            channel=0,
-                            language='English',
-                            speaker=re.sub(r'-.*', r'', idx),
-                            text=subseg_info.text
-                        ))
+        for recording in audio:
+            for source in recording.sources:
+                # In AMI "source.channels" will always be a one-element list
+                channel = source.channels[0]
```
[the remark this answers is not in the document]
Comment (pzelasko) [reply lhotse/recipes/ami.py:215]: nice trick

**Comment by pzelasko** [comment]
I extended the PR to make it even easier to work with multi-channel cuts - now the `load_features` and `load_audio` methods on `MixedCut` can return either the mono-mix, or all the component tracks just before mixing, i.e. with offsets and SNRs applied to them, if they are defined.

I checked how it works on some AMI recordings (the `plot_tracks_audio/features` method use the updated mixing mechanism):

image

(BTW the difference in feature-domain mix is quite noticable from the time-domain mix when using 4 channels, but this is not likely that we're going to use it this way, rather a sanity check if all works OK in addition to unit tests)

It's ready for a final review and merge.

**Comment by danpovey** [comment]
Great!

**Pull request merged**